### PR TITLE
fix(chat): add spinner while create new entity or delete all entities (Issue #725)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
- Issue: https://github.com/epam/ai-dial-chat/issues/725

### CODE REVIEW

**Comment by ai-dial-actions** [comment]
Hey there and thank you for opening this pull request! 👋🏼

We require pull request titles to follow the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/) and it looks like your proposed title needs to be adjusted.

Details:

```
No release type found in pull request title "Fix/725 new entity spinner". Add a prefix to indicate what kind of release this pull request corresponds to. For reference, see https://www.conventionalcommits.org/

Available types:
 - fix
 - hotfix
 - feat
 - feature
 - chore
```

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-765.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-765.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chatbar/ChatbarSettings.tsx [hunk 1]
```diff
@@ -52,14 +52,19 @@ export const ChatbarSettings = () => {
   const maximumAttachmentsAmount = useAppSelector(
     ConversationsSelectors.selectMaximumAttachmentsAmount,
   );
+  const isActiveNewConversationRequest = useAppSelector(
+    ConversationsSelectors.selectIsActiveNewConversationRequest,
+  );
 
   const handleToggleCompare = useCallback(() => {
+    if (isActiveNewConversationRequest) return;
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chatbar/ChatbarSettings.tsx]: use `disabled` option instead
```
disabled: isStreaming || isActiveNewConversationRequest,
```

File: apps/chat/src/components/Chatbar/Chatbar.tsx [hunk 2]
```diff
@@ -33,23 +35,31 @@ const ChatActionsBlock = () => {
...
+  const isActiveNewConversationRequest = useAppSelector(
+    ConversationsSelectors.selectIsActiveNewConversationRequest,
+  );
 
   return (
     <div className="flex px-2 py-1">
       <button
         className="flex shrink-0 grow cursor-pointer select-none items-center gap-3 rounded px-3 py-2 transition-colors duration-200 hover:bg-accent-primary-alpha disabled:cursor-not-allowed"
         onClick={() => {
+          dispatch(ConversationsActions.setIsActiveConversationRequest(true));
```
Comment (mikitabut) [inline apps/chat/src/components/Chatbar/Chatbar.tsx]: Why are you doing this here?
Do it in reducer

File: apps/chat/src/components/Chatbar/ChatbarSettings.tsx [hunk 3]
```diff
@@ -52,8 +52,12 @@ export const ChatbarSettings = () => {
   const maximumAttachmentsAmount = useAppSelector(
     ConversationsSelectors.selectMaximumAttachmentsAmount,
   );
+  const isActiveNewConversationRequest = useAppSelector(
+    ConversationsSelectors.selectIsActiveNewConversationRequest,
+  );
 
   const handleToggleCompare = useCallback(() => {
+    dispatch(ConversationsActions.setIsActiveConversationRequest(true));
```
Comment (mikitabut) [inline apps/chat/src/components/Chatbar/ChatbarSettings.tsx]: Same

**Review by denys-kolomiitsev — DISMISSED** [review]

File: apps/chat/src/components/Header/CreateNewChatMobile.tsx [hunk 4]
```diff
@@ -18,23 +23,36 @@ export const CreateNewChatMobile = ({ iconSize }: Props) => {
...
+    ConversationsSelectors.selectIsActiveNewConversationRequest,
+  );
+
   return (
     <Tooltip isTriggerClickable tooltip={t('New conversation')}>
       <div
         className="flex h-full items-center justify-center border-r border-tertiary px-3 md:hidden"
         onClick={() => {
+          if (!isConversationsLoaded || isActiveNewConversationRequest) return;
+          dispatch(ConversationsActions.setIsActiveConversationRequest(true));
```
Comment (mikitabut) [inline apps/chat/src/components/Header/CreateNewChatMobile.tsx]: same

File: apps/chat/src/components/Promptbar/Promptbar.tsx [hunk 5]
```diff
@@ -30,18 +31,28 @@ const PromptActionsBlock = () => {
...
+  const isPromptRequestSent = useAppSelector(
+    PromptsSelectors.selectIsActiveNewPromptRequest,
+  );
+
   return (
     <div className="flex px-2 py-1">
       <button
         className="flex shrink-0 grow cursor-pointer select-none items-center gap-3 rounded px-3 py-2 transition-colors duration-200 hover:bg-accent-primary-alpha disabled:cursor-not-allowed"
         onClick={() => {
+          dispatch(PromptsActions.setIsActiveNewPromptRequest(true));
```
Comment (mikitabut) [inline apps/chat/src/components/Promptbar/Promptbar.tsx]: same

File: apps/chat/src/store/prompts/prompts.epics.ts [hunk 6]
```diff
@@ -89,16 +89,22 @@ const createNewPromptEpic: AppEpic = (action$, state$) =>
...
-                'An error occurred while creating a new prompt. Most likely the prompt already exists. Please refresh the page.',
+          return concat(
+            of(
+              UIActions.showErrorToast(
+                translate(
+                  'An error occurred while creating a new prompt. Most likely the prompt already exists. Please refresh the page.',
+                ),
               ),
             ),
+            of(PromptsActions.setIsActiveNewPromptRequest(false)),
```
Comment (mikitabut) [inline apps/chat/src/store/prompts/prompts.epics.ts:104]: looks like finalize is better place
or reducer for createNewPromptSuccess

File: apps/chat/src/store/prompts/prompts.reducers.ts [hunk 7]
```diff
@@ -230,14 +236,19 @@ export const promptsSlice = createSlice({
       });
     },
     duplicatePrompt: (state, _action: PayloadAction<PromptInfo>) => state,
-    setPrompts: (state, { payload }: PayloadAction<{ prompts: Prompt[] }>) => {
-      state.prompts = payload.prompts;
+    setPrompts: (
+      state,
+      { payload }: PayloadAction<{ prompts: PromptInfo[] }>,
```
Comment (mikitabut) [inline apps/chat/src/store/prompts/prompts.reducers.ts]: add flag as here and update share epics for using it
https://github.com/epam/ai-dial-chat/pull/781/files#diff-be8fb500bbbc8a5659fc578fd9bc33f4c614b0a343a560b30c7457de12b203f7R348

**Review by mikitabut — APPROVED** [review]

**Pull request merged**